addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -163,13 +163,24 @@ export class LinkPlugin extends Plugin {
                 description: _t("Remove Link"),
                 icon: "fa-unlink",
                 isAvailable: (selection) => {
-                    const linkEl = findInSelection(selection, "a");
-                    return (
-                        !!linkEl &&
-                        !this.isLinkImmutable(linkEl) &&
-                        !this.isUnremovable(linkEl) &&
-                        isHtmlContentSupported(selection)
-                    );
+                    const commonElementAncestor = closestElement(selection.commonAncestorContainer);
+                    if (!commonElementAncestor) {
+                        return false;
+                    }
+                    for (const linkEl of [
+                        ...selectElements(commonElementAncestor, "a"),
+                        commonElementAncestor.closest("a"),
+                    ]) {
+                        if (
+                            linkEl &&
+                            selection.intersectsNode(linkEl) &&
+                            !this.isLinkImmutable(linkEl) &&
+                            !this.isUnremovable(linkEl) &&
```

Comment:
I'm thinking about an edge case where we have a unremovable link which has more than one character, e.g. `<a class="oe_unremovable" href="http://test.test">ddd[dd]</a>`. Should we enable the unlink for this partial selection?

Cause we can partially unlink it when there's another removable link in the selection `c<a class="oe_unremovable" href="http://test.test">dd[ddd</a>e<a href="http://test.test">f]</a></p>`. 

So we can also check if the unremoveable link is fully selected :thinking: (I hope it won't break other edge cases)